Lib/test/test_compileall.py

```diff
@@ -200,7 +200,10 @@ class CompileallTestsWithSourceEpoch(CompileallTestsBase,
                                      unittest.TestCase,
                                      metaclass=SourceDateEpochTestMeta,
                                      source_date_epoch=True):
-    pass
+    def test_compile_all_2038(self):
+        with open(self.source_path, 'r') as f:
+            os.utime(f.name, (2147558400, 2147558400)) # Jan 20, 2038 as touch
+        self.assertTrue(compileall.compile_file(pathlib.Path(self.source_path)))
```

Comment:
not sure, but maybe this is not a good place in this file to add the test in, because SOURCE_DATE_EPOCH changes the default .pyc type to be hash-based, which would not hit the error condition. See #9607

Reply:
It should be moved to CompileallTestsBase.

Reply:
done